build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -157,7 +158,22 @@ private static void RemovePdbsFromDir(string directory)
             FS.RmFilesInDirRecursive(directory, "*.pdb");
         }
 
-        private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli dotnet, string rootOutputDirectory)
+        private static BuildTargetResult CompileCliSdkAndGenerateNuGetPackagesArchive(
+            BuildTargetContext c,
+            DotNetCli dotnet,
+            string rootOutputDirectory,
+            DotNetCli currentDotnet)
```

Comment:
Not true. Look above. Dotnet is Stage1 but the current is Stage2. Technically, we could use the same, but I want the dotnet new that we are going to run to be ran using the stage in which we are going to place archive.